plugins/discovery/config_test.go

```diff
@@ -50,11 +52,22 @@ func TestConfigValidation(t *testing.T) {
 			services: []string{"service1"},
 			wantErr:  false,
 		},
+		{
+			input:    `{"name": "a/b/c", "decision": "query", "signing": {"keyid": "foo", "scope": "write"}}}`,
+			services: []string{"s1"},
+			wantErr:  false,
+		},
+		{
+			input:    `{"name": "a/b/c", "decision": "query", "signing": {"keyid": "bar", "scope": "write"}}}`,
```

Comment:
We can add a check to see if the key that discovery depends on has changed like we do for services. 

The `keyid` could be included in the bundle signature, which we could read from the bundle itself. Since we only support one signature in the bundle,  this would work.

If the `keyid` was not provided in the signature, we could pick it up from `discovery.signing`.

WDYT ?

Reply:
Yep, except per the offline discussion yesterday, we should make the config keyid override the bundle keyid.

Reply:
Added a check to avoid updates to keys in the boot config.